hs_core/views/utils.py

```diff
@@ -1072,10 +1072,16 @@ def unzip_file(user, res_id, zip_with_rel_path, bool_remove_original,
     try:
         # unzip to a temporary folder
         unzip_path = istorage.unzip(zip_with_full_path, unzipped_folder=uuid4().hex)
+        dir_file_list = istorage.listdir(unzip_path)
+        unzip_subdir_list = dir_file_list[0]
         # list all files to be moved into the resource
         unzipped_files = listfiles_recursively(istorage, unzip_path)
         unzipped_foldername = os.path.basename(unzip_path)
         override_tgt_paths = []
+        for sub_dir_name in unzip_subdir_list:
+            dest_sub_dir_path = os.path.join(os.path.dirname(zip_with_full_path), sub_dir_name)
+            if istorage.exists(dest_sub_dir_path):
+                override_tgt_paths.append(dest_sub_dir_path)
```

Comment:
I think we may have to have separate list for folders that will be overridden. That way we can first loop though that folder list first and delete folders before deleting the files that would be overridden.

Reply:
I tried to make changes as you suggested since your comment made sense to me. However, after doing so, the bug was not fixed any more. After debugging, I realized the issue was related to a folder overriding a file with the same name, so we cannot really use the list for folders for overriding to determine whether we need to remove folder or file from the destination folder due to the fact that a folder with the same name as a file in the destination folder cannot be moved to the destination folder until the file with the same name is removed. The reverse is true when moving a file to a folder with the same name, so the user has to determine whether they want to really override either files or folders with the same name as the files or folders to be unzipped. It is really confusing, but I have to go back to the old logic without splitting into folders and files to get this working.